src/lib/structures/symbols.mli

```diff
@@ -128,9 +128,23 @@ val print_clean : Format.formatter -> t -> unit
 
 (*val dummy : t*)
 
-val fresh : ?is_var:bool -> string -> t
+module type Id = sig
+  val fresh : ?base:string -> unit -> string
+  val reset_fresh_cpt : unit -> unit
+  val is_id : string -> bool
+  val make_as_fresh : string -> string
+end
+
+module InternalNameId : Id
+module SkolemId : Id
+
+val fresh_name : unit -> t
+val is_internal_name : t -> bool
+
+val fresh_skolem : ?is_var:bool -> string -> t
+val make_as_fresh_skolem : string -> t
+val is_skolem : t -> bool
```

Comment:
Suggestion, just for consistency:
`fresh_name` -> `fresh_interna_name`
`is_internal_name` -> `is_fresh_internal_name`
`is_skolem` -> `is_fresh_skolem`